model_compression_toolkit/core/keras/graph_substitutions/substitutions/dwconv_to_conv.py

```diff
@@ -0,0 +1,108 @@
+from model_compression_toolkit.constants import REUSE, REUSE_GROUP
+from model_compression_toolkit.core.keras.reader.node_builder import REUSED_IDENTIFIER
+from model_compression_toolkit.core.keras.constants import KERNEL, BIAS, USE_BIAS, NUM_HEADS, KEY_DIM, VALUE_DIM, \
+    QUERY_SHAPE, KEY_SHAPE, VALUE_SHAPE, OUTPUT_SHAPE, ATTENTION_AXES, ACTIVATION, GROUPS, LINEAR, FILTERS, PADDING, \
+    FUNCTION, DIMS, TARGET_SHAPE, F_STRIDED_SLICE, F_STACK, Q_KERNEL, Q_BIAS, K_KERNEL, K_BIAS, V_KERNEL, V_BIAS, \
+    OUTPUT_KERNEL, OUTPUT_BIAS, F_MATMUL, TRANSPOSE_B, KERNEL_SIZE, AXIS, F_STRIDED_SLICE_BEGIN, F_STRIDED_SLICE_END, \
+    STRIDES, DILATIONS
+
+
+class DwconvToConv(common.BaseSubstitution):
+    """
+    Replace DepthWise Conv2d layer into a Conv2d layer with number of groups equal the number of input channel
+    """
+
+    def __init__(self):
+        """
+        Matches DepthwiseConv2D node.
+        """
+        super().__init__(matcher_instance=NodeOperationMatcher(DepthwiseConv2D))
+
+    @staticmethod
+    def _get_weight_by_name(mha_node, w_str):
+        w = [k for k in mha_node.weights.keys() if w_str in k]
+        return w[0] if w else None
+
+    def substitute(self,
+                   graph: Graph,
+                   dwconv_node: BaseNode) -> Graph:
+        """
+        Replace DepthwiseConv2D to Conv2D (with groups = #output_channels) for regularity of MCT quantization
+        parameters.
+        Args:
+            graph: Graph we apply the substitution on.
+            dw_conv_node: DepthwiseConv2D node to replace.
+        Returns:
+            Graph after applying the substitution.
+        """
+
+        # Skip in case mult depth_multiplier=1
+        if dwconv_node.framework_attr['depth_multiplier'] == 1:
+            return graph
+
+        k = dwconv_node.weights[self._get_weight_by_name(dwconv_node, KERNEL)].copy()
+        k_shape = k.shape
+        filters = k_shape[2] * k_shape[3]
+        k = np.reshape(k,[k_shape[0], k_shape[1], 1, k_shape[2] * k_shape[3]])
+        bias_key = self._get_weight_by_name(dwconv_node, BIAS)
+        _reuse_params = {REUSE: dwconv_node.reuse, REUSE_GROUP: dwconv_node.reuse_group}
+
+        dwconv_fw_attr = {FILTERS: filters, KERNEL_SIZE: k_shape[0],
+                                                    STRIDES: dwconv_node.framework_attr[STRIDES],
+                                                    PADDING: dwconv_node.framework_attr[PADDING],
+                                                    DILATIONS: dwconv_node.framework_attr[DILATIONS],
+                                                    GROUPS: k_shape[2],
+                                                    ACTIVATION: dwconv_node.framework_attr[ACTIVATION],
+                                                    USE_BIAS: dwconv_node.framework_attr[USE_BIAS]}
+        if bias_key:
+            b = dwconv_node.weights[bias_key].copy()
```

Comment:
You can rewrite this if-else simpler (create a dictionary of the weights updated with the bias if there's such and initiate the BaseNode. This way, the duplication of the node initiation can be avoided).